src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -148,18 +148,27 @@ static int32_t peep_move_one_tile(Direction direction, Peep* peep)
         //
         // What we want instead is to apply that randomness in the direction they are walking ONLY, and keep their
         // other coordinate constant.
+        //
+        // However, we have also seen some situations where guests end up too far from the center of paths. We've
+        // not identified exactly what causes this yet, but to limit the impact of it, we don't just keep the other
```

Comment:
Isn't it just when the destination tolerance is cranked up to a large value like during aimless wandering mode.